### DISCUSSION TITLE
Model Parameters and Lammps Performance

### DISCUSSION THREAD

**Discussion by owen-rett** [ACEsuit/mace#132] [Q&A]
Hello, 
I'm in the process of learning how to use MACE, and have the CPU version of lammps installed, using the dev branch of mace to train the model, and I was wondering what parameters in model training will impact performance the most, or at least some settings that should be worth experimenting with. I'm currently doing some initial performance testing using a very small (192 atom) unit cell, and a L=1 model, and am seeing 1.535 timesteps/s on 40 cores and one node (lammps is reporting 2719% CPU usage), which makes me suspect I can probably find some increase to speed. I've not checked the GPU version of mace-lammps yet, as there's an issue with the cmake version on the cluster I'm using, but will check it soon. I've attached the current shell script I'm using to call the mace training. 
[mace_train.txt](https://github.com/ACEsuit/mace/files/12062771/mace_train.txt)

Also, with the current lammps interface, does the choice of default_dtype= float32 vs. float64 during model training affect inference speed?

**Comment by davkovacs** [comment]
Hi! 
I recommend looking at this paper to get a sense for how various parameters affect the performance of the model. 
I would say increasing L is probably the main contributor to increased computational cost, so it might be a good idea to try an L=0 model if you are concerned with computational speed. The number of channels is the other one, 128 is a good average size, but you can try smaller and you will see an increase in speed. 
As for floating point precision you will find that float32 is typically faster than using float64 precision. 

Also note, that MACE is primarily a GPU code, and the CPU evaluation speeds are *much* slower than the GPU speeds, so I definitely recommend fixing that if possible. 

Otherwise your initial fitting script looks sensible to me.

**Comment by owen-rett** [comment]
Ok, thanks for the response. I've gotten the GPU version of lammps working, and it seems to be running well. I'll try out L=0,1,2 models to check system size / memory requirements.
